### ISSUE TITLE
Collect Chart version metadata

### ISSUE THREAD

**Issue by arbourd** [linkyard/concourse-helm-resource#67]
We want to start collecting the Chart version (and perhaps the App version) metadata from this resource. We regularly deploy charts that are "latest" but would like a way to track what version we just deployed.

Here's the general idea:

1. `helm repo update`
1. `helm search repo/chart` if Chart is being deployed _without_ a version
1. Collect Chart version
1. `helm upgrade -i repo/chart` Normal upgrade/install path
1. Include deployed Chart version in metadata on success

Is this something that you would accept a PR for?

Thanks!

**Comment by msiegenthaler** [comment]
Wouldn't it be better to model the chart as a resource (check/get) and then deploy from that?
We're currently missing the check/get functionality in this resource and I'm considering adding a second resource "helm-chart" that would be able the check/get/put helm charts from a helm repo ("native" or S3). The helm-chart resource would download the chart to a .tar.gz and you could then do a put using the file and the concourse-helm-resource. That way you'd have full transparency (and control) over the deployed chart version.
What do you think about this proposal?

**Comment by arbourd** [comment]
I think that is definitely a better model of deployment and I was actually thinking of creating a resource like that, heavily based upon this one as well (if I get there first I can donate it to you as you'd probably be better stewards).

Although, I'd like to know what this Helm resource is deploying. I believe this should actually be reported by Helm itself (why not right?), but it isn't at this time. I wonder if there's a better way to get that metadata...

TL;DR: still love to see it here
<img width="315" alt="screen shot 2018-05-28 at 11 53 58 am" src="https://user-images.githubusercontent.com/7211830/40622324-e08b0092-626e-11e8-9fbd-9aa421d313d1.png">

**Comment by arbourd** [comment]
> We're currently missing the check/get functionality in this resource and I'm considering adding a second resource "helm-chart" that would be able the check/get/put helm charts from a helm repo ("native" or S3)

Hey @msiegenthaler I put together https://github.com/arbourd/helm-chart-resource specifically for this reason. I'd love to transfer ownership of it to Linkyard (I feel like you'd be great stewards of it) and continue to hack on it.

Are you interested?

**Comment by msiegenthaler** [comment]
Wow, that looks great!
Sure, we'd be glad to take the ownership of it.

**Comment by arbourd** [comment]
Awesome! I've sent the transfer to you directly (because I can't send it to orgs). 🤓

**Comment by msiegenthaler** [comment]
Great, that worked fine. I also set up a Dockerhub build, so the image should now be available as `linkyard/helm-chart-resource` on Dockerhub.
I've also created a 0.1.0 version so we can test if everything works.
